### ISSUE TITLE
Gosia has single-precision constants in double-precision datasets

### ISSUE THREAD

**Issue by m1el** [warr/gosia#1]
For example, [eng](https://github.com/warr/gosia/blob/gosia_20110524/gosia.f#L181) is defined as double-precision float (`REAL*8`), but [data section](https://github.com/warr/gosia/blob/gosia_20110524/gosia.f#L308) sets single-precision values to it (without `D0` suffix).

As a result, `gdet_test` relies on this, and if you run `generate_gdet.inp` with double precision output, you will get first energy as `5.0000000745058060E-002`. Which is incorrect and should be `5.0000000000000003E-002`.

This probably has some other ramifications, e.g. all of the calculations performed in double-precision, when all of the precision is lost due to single-precision constants [like here](https://github.com/warr/gosia/blob/gosia_20110524/ylm.f#L53).

**Comment by warr** [comment]
> For example, [eng](https://github.com/warr/gosia/blob/gosia_20110524/gosia.f#L181) is defined as double-precision float (`REAL*8`), but [data section](https://github.com/warr/gosia/blob/gosia_20110524/gosia.f#L308) sets single-precision values to it (without `D0` suffix).
>
> As a result, `gdet_test` relies on this, and if you run `generate_gdet.inp` with double precision output, you will get first energy as `5.0000000745058060E-002`. Which is incorrect and should be `5.0000000000000003E-002`.
>
> This probably has some other ramifications, e.g. all of the calculations performed in double-precision, when all of the precision is lost due to single-precision constants [like here](https://github.com/warr/gosia/blob/gosia_20110524/ylm.f#L53).

Indeed, there is some drop in precision of constants. This should, however,
still be small compared to the intrinsic accuracy of the whole Coulomb
excitation method, which entails quite a lot of approximations. Whilst such
numeric differences can be annoying, they shouldn't change the physics. I also
note that Gosia is full of hard-coded constants, some of which are not
completely accurate, but "good enough" for the purpose. (e.g. the attenuation
coefficients for Ge differ a little from the NIST values). The meaning of some
of these constants was lost, when the original author passed away in 2006. So
it would be quite a job to find and correct them all and the difference should
be less than the error due to the physics approximations. The double precision
is really only needed to reduce the successive accumulation of floating point
error when performing tasks such as integration.

So I agree, this _should_ be fixed, but it is a big job and I don't think it
will change the physics. I'll add it to the "to do" list.

-- 
Dr. Nigel Warr, Institut für Kernphysik, Universität zu Köln,
Zülpicherstr. 77, D-50937 Köln, Germany. Tel +49 221 470 3459
